src/workersrc.js

```diff
@@ -83,9 +87,56 @@ function initializeCustomPDFViewerApplication() {
 		logger.info('Download, print and user interaction disabled')
 	} else {
 		logger.info('Download and print available')
+
+		PDFViewerApplication.download = function() {
+			alert(t('files_pdfviewer', 'Please make some changes and click save.'))
+		}
+
+		PDFViewerApplication.save = function() {
+			logger.info('PDF Document with annotation is being saved.')
+			PDFViewerApplication.pdfDocument.saveDocument().then(function(data) {
+				if (isUploading) {
+					alert(t('files_pdfviewer', 'Please wait for the upload to complete.'))
+					return
+				}
+
+				savePdf(data, PDFViewerApplication)
+			})
+		}
 	}
 
 	logger.debug('Initialized files_pdfviewer', PDFViewerApplicationOptions.getAll())
 }
 
+function savePdf(data, pdfAppication) {
+	isUploading = true
+	uploadProgressMessage.style.display = 'block'
+	const blob = new Blob([data], { type: 'application/pdf' })
+
+	// Create a URL object and obtain file query parameter.
+	const parsedURL = new URL(location.href)
+	const filePath = parsedURL.searchParams.get('file')
+
+	// Access the OC object in the parent window
+	const parentWindow = window.parent
+	const requestToken = parentWindow.OC.requestToken
+
+	// Uploading file with nextcloud axios. This will create a new file version if
+	// versions app is installed.
+	axios.put(filePath, blob, {
+		headers: {
+			'Content-Type': 'application/pdf',
+			requesttoken: requestToken,
+		},
+	}).then(function() {
+		logger.log('File uploaded successfully')
+		isUploading = false
+		uploadProgressMessage.style.display = 'none'
+	}).catch(function(error) {
+		alert(t('files_pdfviewer', 'File upload failed.'))
```

Comment:
Instead of alerts it would be better to use, for example, toasts, as they are less... _disrupting_? _aggressive_? I do not know how to say it, but I hope it is understandable :-)

This could be done with https://github.com/nextcloud-libraries/nextcloud-dialogs:
```
import { showError } from '@nextcloud/dialogs'

[...]

showError(t('files_pdfviewer', 'File upload failed.'))
```